src/dlnodesession.cpp

```diff
@@ -172,7 +172,7 @@ Status DLNodeSession::validate(const std::shared_ptr<ov::runtime::Tensor>& tenso
     if (!tensorInfo.getShape()[batchIndex.value()].match(dims[batchIndex.value()])) {
         // If remaining dimensions are equal, it is invalid batch size
         std::stringstream ss;
-        if (!tensorInfo.getShape().match(dims, batchIndex.value())) {
+        if (tensorInfo.getShape().match(dims, batchIndex.value())) {
```

Comment:
No, we first check in L172 if the batch size is wrong, if so, we proceed with checking if remaining dimensions are OK (L175). If so, we report INVALID_BATCH_SIZE, otherwise we report INVALID_SHAPE. This change is required because the unit tests found out the logic before change was wrong.

Reply:
OK it seems that the implementation was changed with previous MR and after now you fix this here.